tests/ordered-elements_test.js

```diff
@@ -53,6 +53,38 @@ describe("OrderedElements", () => {
         }, elems);
     });
 
+    it('moves overridden elements to the end', () => {
+        const elems = new OrderedElements();
+
+        elems.set("a", 1);
+        elems.set("b", 1);
+        elems.set("a", 2);
+
+        assert.deepEqual({
+            elements: {
+                b: 1,
+                a: 2,
+            },
+            keyOrder: ["b", "a"],
+        }, elems);
+    });
+
+    it('can preserve order when setting', () => {
+        const elems = new OrderedElements();
+
+        elems.set("a", 1);
+        elems.set("b", 1);
+        elems.set("a", 2, { preserveOrder: true });
```

Comment:
this isn't how the API works, right? It should just be a `true` here

Reply:
Oh, right. This was an earlier revision. I'll fix now.